sim_pipeline/gg_lens.py

```diff
@@ -81,15 +81,24 @@ def validity_test(self, min_image_separation=0, max_image_separation=10):
 
         kwargs_lens = kwargs_params['kwargs_lens']
         image_positions = lens_eq_solver.image_position_from_source(source_pos_x, source_pos_y, kwargs_lens)
-        if len(image_positions) < 2:
+        if len(image_positions[0]) < 2:
             return False
 
+        if len(image_positions[0]) == 2:
+            max_image_separation = np.sqrt((image_positions[0][0] - image_positions[0][1]) ** 2 + (
+                        image_positions[1][0] - image_positions[1][1]) ** 2)
+        else:
+            coords = np.stack((image_positions[0], image_positions[1]), axis=-1)
+            separations = np.sqrt(np.sum((coords[:, np.newaxis] - coords[np.newaxis, :]) ** 2, axis=-1))
+            max_image_separation = np.max(separations)
+        if max_image_separation < min_image_separation:
+            return False
+        if max_image_separation > max_image_separation:
```

Comment:
I am not sure this line makes much sense

Reply:
yeah, this is a typo, thank you so much!